consensus/pacemaker.go

```diff
@@ -154,42 +156,41 @@ func (p *paceMaker) RestartTimer() {
 	if p.stepCancelFunc != nil {
 		p.stepCancelFunc()
 	}
-	p.debugSleep()
 
 	// NOTE: Not defering a cancel call because this function is asynchronous.
+
 	stepTimeout := p.getStepTimeout(p.consensusMod.Round)
-	ctx, cancel := context.WithTimeout(context.TODO(), stepTimeout)
+
+	clock := p.bus.GetClock()
+
+	ctx, cancel := clock.WithTimeout(context.TODO(), stepTimeout)
 	p.stepCancelFunc = cancel
 
 	go func() {
 		select {
 		case <-ctx.Done():
 			if ctx.Err() == context.DeadlineExceeded {
-				p.consensusMod.nodeLog(typesCons.PacemakerTimeout(p.consensusMod.Height, p.consensusMod.Step, p.consensusMod.Round))
+				p.consensusMod.nodeLog(typesCons.PacemakerTimeout(p.consensusMod.CurrentHeight(), p.consensusMod.Step, p.consensusMod.Round))
 				p.InterruptRound()
 			}
-		case <-time.After(stepTimeout + 30*time.Millisecond): // Adding 30ms to the context timeout to avoid race condition.
+		case <-clock.After(stepTimeout + 30*timePkg.Millisecond): // Adding 30ms to the context timeout to avoid race condition.
 			return
 		}
 	}()
 }
 
 func (p *paceMaker) InterruptRound() {
-	p.consensusMod.nodeLog(typesCons.PacemakerInterrupt(p.consensusMod.Height, p.consensusMod.Step, p.consensusMod.Round))
+	p.consensusMod.nodeLog(typesCons.PacemakerInterrupt(p.consensusMod.CurrentHeight(), p.consensusMod.Step, p.consensusMod.Round))
```

Comment:
I did a grep for `.Lock()` and can see that literally we've isolated it to two places:
1. HandleMessage
2. HandleDebugMessage

This makes sense and I like because it is the entrypoint to all the other business logic. 

The only piece I believe it's missing (though I can't say I'm 100% confident) is asynchronous events, whereas the `HandleMessage` and `HandleDebugMessage` are synchronous in their downstream operations.

What I did is grep for `go ` in the rest of the consensusModule and `RestartTimer` is the only thing that came up.

tl;dr I think we need to acquire a lock at the beginning of `InterruptRound` as well.

Reply:
`InterruptRound` is called internally when we handle messages, locking in there, even if it sounds correct, would cause deadlocks.

I have added some info about this here https://github.com/pokt-network/pocket/pull/198/commits/46b1d648b52e445c09d228760df10363a8fff1c0

Reply:
LGTM!